### PR TITLE
Implement striding and aggregation for Swath- and AreaDefinitions

### PR DESCRIPTION
**Pull request by mraspaud**
This PR implements striding and aggregation for a couple of Definition objects.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

File: pyresample/geometry.py [hunk 1]
```diff
@@ -547,6 +547,39 @@ def __init__(self, lons, lats, nprocs=1):
         elif lons.ndim > 2:
             raise ValueError('Only 1 and 2 dimensional swaths are allowed')
 
+    def copy(self):
+        return SwathDefinition(self.lons, self.lats)
```
Comment (djhoese) [inline pyresample/geometry.py]: Shouldn't there be a check and special handling of numpy arrays?

File: pyresample/geometry.py [hunk 2]
```diff
@@ -547,6 +547,39 @@ def __init__(self, lons, lats, nprocs=1):
...
+
+    def aggregate(self, **dims):
```
Comment (djhoese) [inline pyresample/geometry.py]: Docstring? And is this another function that won't work in python 2?

File: pyresample/geometry.py [hunk 3]
```diff
@@ -906,6 +938,29 @@ def __init__(self, area_id, description, proj_id, projection, width, height,
...
+        """Return an aggregated version of the area."""
+        width = int(self.width / dims.get('x', 1))
+        height = int(self.height / dims.get('y', 1))
+        return self.copy(height=height, width=width)
+
+    @property
+    def shape(self):
+        return (self.height, self.width)
```
Comment (djhoese) [inline pyresample/geometry.py]: Unnecessary parentheses.

File: pyresample/geometry.py (hunk 1 above)
Comment (mraspaud) [reply pyresample/geometry.py]: That should be done in the constructor then

File: pyresample/geometry.py (hunk 2 above)
Comment (mraspaud) [reply pyresample/geometry.py]: The aggregation in this case doesn't need xarray's coarsen, so it should work with python 2 too.

Comment (mraspaud) [reply pyresample/geometry.py]: Fixed

File: pyresample/geometry.py (hunk 3 above)
Comment (mraspaud) [reply pyresample/geometry.py]: Fixed

File: pyresample/geometry.py (hunk 1 above)
Comment (mraspaud) [reply pyresample/geometry.py]: And it is

File: pyresample/geometry.py [hunk 4]
```diff
@@ -547,6 +547,48 @@ def __init__(self, lons, lats, nprocs=1):
...
+            return np.dstack((x, y, z))
+
+        geocent = pyproj.Proj(proj='geocent')
+        latlong = pyproj.Proj(proj='latlong')
+        res = da.map_blocks(do_transform, latlong, geocent,
+                            self.lons.data, self.lats.data,
+                            da.zeros_like(self.lons.data), new_axis=[2],
+                            chunks=(self.lons.chunks[0], self.lons.chunks[1], 3))
+        res = DataArray(res, dims=['y', 'x', 'coord'], coords=self.lons.coords)
+        res = res.coarsen(**dims).mean()
```
Comment (djhoese) [inline pyresample/geometry.py]: You said it doesn't need `coarsen` but it is using it right here. Right?

Comment (mraspaud) [reply pyresample/geometry.py]: This one does, not the AreaDefinition one

**Pull request merged**